R/functions.R

```diff
@@ -4001,23 +4001,43 @@ chmAddTSNE <- function(hm, axis, tsne, pointIds, basename = "TSNE") {
 #' @seealso [chmAddReducedDim()]
 
 chmAddPCA <- function(hm, axis, prc, basename = "PC", ndim = 2) {
-  stopifnot(is(hm, "ngchmVersion2"))
-  stopifnot(mode(axis) == "character" && length(axis) == 1)
-  stopifnot(axis == "row" || axis == "column")
-  stopifnot(mode(basename) == "character" && length(basename) == 1)
-  stopifnot(mode(ndim) == "numeric" && length(basename) == 1)
+  if (class(hm) != "ngchmVersion2") {
+    stop("First argument (hm) must be an ngchmVersion2 object.")
+  }
+  if (mode(axis) != "character" || length(axis) != 1) {
```

Comment:
Trivial, but -
I would combine these two tests then line 4011 would be the appropriate (and more informative) error message for both.   For that matter, if it's not 'row' or 'column' do we even need to check its mode or length?

Reply:
Line 4011 is the best error message.
R can do funny things if the argument to the test is not a single string.  I would rather we give an error message in that case, rather than relying on whatever error R eventually throws.
If there is a convenient way to way to show the user both error messages in the case the second argument is not a single string that would be best.
Also good would outputting some information about the value of axis when there is an error. Easy if it is a single string. More tedious if it is something else.
Maybe adding a helper function that could be used in similar tests would be good.